src/main/java/com/bootcamp/demo/FirebaseController.java

```diff
@@ -42,16 +53,67 @@ private void initFirestore() throws IOException {
         initializeApp(options);
 
         firestoreDB = getFirestore();
+        bookingService.setDb(firestoreDB);
     }
 
     /**
      * Returns the path for all collections stored in Firestore
      */
     @GetMapping("/getAllPaths")
     public Set<String> getAllPaths() {
-        return StreamSupport.stream(firestoreDB.listCollections().spliterator(), false)
+        return stream(firestoreDB.listCollections().spliterator(), false)
                 .map(CollectionReference::getPath)
                 .collect(Collectors.toUnmodifiableSet());
 
     }
+
+    @PostMapping("/createBooking")
+    public ResponseEntity<Object> createBooking(final Booking booking){
+        try {
+            LOGGER.info("Booking created successfully. Update time: " + bookingService.createBooking(booking));
+            return SUCCESS_RESPONSE;
+        } catch (ExecutionException | InterruptedException | IllegalArgumentException e) {
+            e.printStackTrace();
+            return FAILURE_RESPONSE;
+        }
+    }
+
+    @GetMapping("/getBookingsByUserId")
+    @ResponseBody
+    public ResponseEntity<Set<Booking>> getBookingsByUserId(final UUID userId) {
```

Comment:
For `@GetMapping`, you should query string params/path variables. If you would like to provide this in the body, you should use `@PostMapping`. You can find an overview for this here: https://www.baeldung.com/spring-new-requestmapping-shortcuts